### PR TITLE
Adding send_info action type

### PR DESCRIPTION
**Pull request by whytheplatypus**
This PR ...

## Changes
- Migrate `task_action` type to include a `send_warning` value
- Use `send_warning` for action history in the case where a receiver has no transports defined.

### Testing
- [x] Tested locally?
- [x] Ran `quickTest all`?
- [x] Ran `./prime test` against local Docker ReportStream container?
- [ ] Downloaded a file from `http://localhost:7071/api/download`?
- [x] Added tests?

## Fixes
*List GitHub issues this PR fixes*
- #1973 

Open review questions:
- Any better name suggestions than `send_info`?
- Is the migration kosher?

---

To explore this locally, change the ignore.CSV transport to have a bogus `host`. Update the `maxRetries = 2` in SendFunction.kt and run end to end tests. You may also have to `@Ignore` a couple of unit tests to pass `gradlew package`. This will populate the action table with a series of `send_warning` and then `send_error` pairs.

### CODE REVIEW

**Comment by whytheplatypus** [comment]
I'm getting some failures in `prime test`. The same failures happen against `master` which makes me think it's unrelated to this change. Confirmation would be much appreciated.

File: prime-router/src/main/resources/db/migration/V17__add_send_info_to_action_type.sql [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  Add new actions 'resend' and 'rebatch', and fix the current task table to use it.
+ */
+ALTER TABLE task        ALTER COLUMN next_action TYPE VARCHAR(255);
+ALTER TABLE report_file ALTER COLUMN next_action TYPE VARCHAR(255);
+ALTER TABLE action      ALTER COLUMN action_name TYPE VARCHAR(255);
+DROP TYPE IF EXISTS task_action;
+CREATE TYPE task_action AS ENUM ('receive', 'translate', 'batch', 'send', 'download', 'wipe', 'batch_error', 'send_error', 'wipe_error', 'none', 'resend', 'rebatch', 'token_auth', 'token_error', 'send_info');
+ALTER TABLE task        ALTER COLUMN next_action TYPE task_action USING next_action::task_action;
+ALTER TABLE report_file ALTER COLUMN next_action TYPE task_action USING next_action::task_action;
+ALTER TABLE action      ALTER COLUMN action_name TYPE task_action USING action_name::task_action;
```
Comment (ronaldheft-usds) [inline prime-router/src/main/resources/db/migration/V17__add_send_info_to_action_type.sql:21]: If I understand this migration correctly, it sets the column to a `VARCHAR` so it doesn't error with dropping `task_action` column. After the column is dropped, the ENUM values are updated and then it sets them back.

**Review by ronaldheft-usds — CHANGES_REQUESTED** [review]
The migration looks good. This is a situation where we probably don't gain much splitting the database migration into a separate PR, but the process would be as follows:

* Submit a separate PR with just the migration
* Get the database PR approved, merged, and deployed into production
* Get the code PR approved and merge after the database changes are in production

File: prime-router/src/main/resources/db/migration/V17__add_send_info_to_action_type.sql (hunk 1 above)
Comment (whytheplatypus) [reply prime-router/src/main/resources/db/migration/V17__add_send_info_to_action_type.sql:21]: That's my understanding as well, the code here was stolen from a previous migration.

**Comment by whytheplatypus** [comment]
@ronaldheft-usds are you looking for this PR to be split up or is it ok to leave as is in this case?

**Review by ronaldheft-usds — APPROVED** [review]
You're good to proceed as is.

**Comment by whytheplatypus** [comment]
@jimduff-usds would you mind taking a look to sanity check the changes in the different transport types?

**Review by jimduff-usds — CHANGES_REQUESTED** [review]
@whytheplatypus , I think as written this won't give the desired behavior for retries.  (which is, that each retry is labelled as a send_warning, then the final one is still a send_error)

Due to the way the code is structured, I suggest you just overwrite the send_warning actionName with a send_error actionName right at the moment in the code that figures out if you've run out of retry chances.   eg, add something like this:
```
                // overwrite existing send_warning or send action name, with send_error:
                actionHistory.action.actionName = TaskAction.send_error
```

Here's what it would look like in the code, around line 133 in SendFunction.kt, but I have totally not tested this:

```
                val msg = "All retries failed.  Send Error report for: $reportId to $serviceName"
                // overwrite existing send_warning or send action name, with send_error:
                actionHistory.action.actionName = TaskAction.send_error
                actionHistory.trackActionResult(msg)
                context.logger.info(msg)
                ReportEvent(Event.EventAction.SEND_ERROR, reportId)
```

File: prime-router/src/main/kotlin/transport/SftpTransport.kt [hunk 2]
```diff
@@ -86,7 +86,7 @@ class SftpTransport : ITransport, Logging {
                     "$sftpTransportType (orgService = ${header.receiver?.fullName ?: "null"})" +
                     ", Exception: ${t.localizedMessage}"
             context.logger.warning(msg)
-            actionHistory.setActionType(TaskAction.send_error)
+            actionHistory.setActionType(TaskAction.send_warning)
```
Comment (whytheplatypus) [inline prime-router/src/main/kotlin/transport/SftpTransport.kt:89]: I think these are somewhat redundant now, but I've left the change for clarity.

Comment (jimduff-usds) [reply prime-router/src/main/kotlin/transport/SftpTransport.kt:89]: Thanks, my exact thought.

**Review by jimduff-usds — APPROVED** [review]
Thanks - this and the work Cle did are the clinchers that will allow us to start notifying on errors.

And - the unit tests were already written, you just had to add a line or two to them.  How pleasing (and rare) is that?

**Pull request merged**